axonserver/src/main/java/io/axoniq/axonserver/localstorage/SerializedEvent.java

```diff
@@ -120,10 +133,16 @@ public String getType() {
         return asEvent().getAggregateType();
     }
 
-    public Map<String, Object> getMetaData() {
-        Map<String,Object> metaData = new HashMap<>();
-        asEvent().getMetaDataMap().forEach((key, value) -> metaData.put(key, asObject(value)));
-        return metaData;
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) {
+            return true;
+        }
+        if (o == null || getClass() != o.getClass()) {
+            return false;
+        }
+        SerializedEvent that = (SerializedEvent) o;
+        return Arrays.equals(serializedData(), that.serializedData());
     }
```

Comment:
So every time `equals()` is called, the event is deserialized? On both sides? Does that really save memory?